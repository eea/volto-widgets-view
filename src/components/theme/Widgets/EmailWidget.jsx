import React from 'react';
import cx from 'classnames';

export const EmailWidget = ({ children, className }) => (
  <a
    href={'mailto:' + children}
    className={cx(className, 'email', 'widget')}
    rel="noreferrer"
    target="_blank"
  >
    {children}
  </a>
);
