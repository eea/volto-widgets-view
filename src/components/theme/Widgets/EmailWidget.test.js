import React from 'react';
import renderer from 'react-test-renderer';
import { EmailWidget } from './EmailWidget';

describe('EmailWidget', () => {
  it('renders an empty email view widget component', () => {
    const component = renderer.create(<EmailWidget />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders an email view widget component', () => {
    const component = renderer.create(
      <EmailWidget className="metadata">foo@bar.com</EmailWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
