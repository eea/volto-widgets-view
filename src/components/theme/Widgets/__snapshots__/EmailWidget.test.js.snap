// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailWidget renders an email view widget component 1`] = `
<a
  className="metadata email widget"
  href="mailto:foo@bar.com"
  rel="noreferrer"
  target="_blank"
>
  foo@bar.com
</a>
`;
